content/en/blog/_posts/2018-05-23-Airflow-Kubernetes-Operator.md

```diff
@@ -0,0 +1,139 @@
+---
+layout: blog
+title:  The Kubernetes "Whatever-your-heart-desires" Operator
+date:   Wednesday, May 23rd 2018
+---
+
+## Introduction
+
+Today, we are excited to announce the Kubernetes Airflow Operator; a new mechanism for launching containers using the Kubernetes cloud deployment framework. 
+
+Since its inception, Airflow's greatest strength has been its flexibility. Airflow offers a wide range of native operators for services ranging from Spark and HBase, to Google Cloud Platform (GCP) and Amazon Web Services (AWS) S3. Airflow also offers easy extensibility through its plug-in framework. However, one limitation of the project is that Airflow users are confined to the frameworks and clients that exist on the Airflow worker at the moment of execution. If a user wishes to use a different version of SciPy or test a new deep learning framework, they would need to either launch a new Airflow cluster or risk conflicting with the dependencies of other users' workflows. 
+
+To address this issue, we've utilized Kubernetes to allow users to launch arbitrary Docker containers and configurations. Airflow users can now have full power over their run-time environments, resources, and secrets, basically turning Airflow into an "any job you want" scheduler.
+
+## What Is Airflow?
+
+Apache Airflow is one realization of the DevOps philosophy of "Code As Configuration." Airflow allows users to launch multi-step pipelines using a simple python object DAG (Directed Acyclic Graph). You can define dependencies, programmatically construct complex workflows, and monitor scheduled jobs in an easy to read UI.
+
+<img src="/images/blog/2018-05-23-Airflow-Kubernetes-Operator/2018-05-23-airflow_dags.png" width="85%" alt="Airflow DAGs" />
+<img src="/images/blog/2018-05-23-Airflow-Kubernetes-Operator/2018-05-23-airflow.png" width="85%" alt="Airflow UI" />
+
+ Airflow comes with built-in operators for frameworks like Apache Spark, BigQuery, Hive, and EMR. It also offers a Plugins entrypoint that allows DevOps engineers to develop their own connectors.
+ 
+## What is Kubernetes?
+
+Before we go any further, let's take a moment for a quick overview of Kubernetes. [Kubernetes](https://kubernetes.io/) is an open-source container deployment engine released by Google and open sourced under the Apache 2.0 licence. Based on Google's [Borg](http://blog.kubernetes.io/2015/04/borg-predecessor-to-kubernetes.html), Kubernetes allows for easy deployment of images using a highly flexible API. Using Kubernetes, you can [deploy Spark jobs](https://github.com/apache-spark-on-k8s/spark), launch end-to-end applications, or create multi-framework ETL pipelines using YAML, JSON, Python, Golang, or Java bindings. The Kubernetes API's programatic launching of containers seemed a perfect marriage with Airflow's "code as configuration" philosophy.
```

Comment:
For the k8s.io blog, this section is probably not needed.

Reply:
That's a very good point. My only concern is how to handle potential readers from the airflow/python communities? Perhaps a quick blurb of "If you're visiting from the python/airflow communities and this is your first k8s exposure, checkout these resources"?